packages/nimble-components/src/select/index.ts

```diff
@@ -668,6 +670,17 @@ export class Select extends FormAssociatedSelect implements ErrorPattern {
         }
     }
 
+    /**
+     * @internal
+     */
+    public registerOption(option: ListOption): void {
+        if (this.options.includes(option)) {
+            return;
+        }
+
+        this.options.push(option);
```

Comment:
Is there an ordering assumption with options? If so, how can we guarantee there isn't an issue here?

Reply:
Yeah, this could result in an order difference between the DOM and what the `Select` displays, which could corrupt the state of things. Perhaps the more rigorous approach is for this implementation to do a `querySelectorAll` on its parent, and find where it's supposed to be inserted into the options. I'll look into this. If I can't find a good solution (and verify that things can get in a bad state), the fallback will be to do a conditional (if possible) `querySelectorAll` when we set the value.

Reply:
From offline discussion, there is a temporal time where it is possible for the select.options state to be in an invalid state. i.e. a new option marked selected is added to the middle of the options list. But this does not impact Angular because the current CVA implementation only looks at the value property on options and creates a value using a unique incrementing id. It doesn't care about the selectedIndex reported by the select where the nimble-select can report incorrect values at times.
We will track the nimble temporal state issues in a tech debt issue mostly to document it but it should not cauuse problems for angular.
Also could not think of alternatives that would not be a potential performance concern at the moment. Discussed options were:
- modify select and selectedIndex to be accessors that on access would access all the child options
- in connectedCallback query all child items and set options to current state (for options inserted one after another, likely expensive / redundant work done on every connectedCallback of children, also does not improve the synchronous behavior since it only runs after connect to the dom so at best a partial solution)

Reply:
Created #1915. I also updated `Combobox` to be a `ListOptionOwner`.